parsec/vpmap.c

```diff
@@ -16,233 +16,264 @@
 
 #include "parsec/parsec_hwloc.h"
 #include "parsec/vpmap.h"
+#include "parsec/parsec_internal.h"
 #include "parsec/utils/debug.h"
 #include "parsec/utils/output.h"
 #include "parsec/constants.h"
 
 /* If HWLOC is not available support up to 64 cores */
 #define DEFAULT_NB_CORE 64
 /**
- * These structures are used by the from_hardware and from_file
- * to store the whole vp map
+ * Structures defining the virtual process (VP) maps. If HWLOC is not
+ * available we only support a single VP, flat across all the available
+ * cores.
  */
 typedef struct {
     int nbcores;
-    int cores[1];
-    int ht[1];
+    int ht;
+    hwloc_cpuset_t cpuset;
 } vpmap_thread_t;
 
 typedef struct {
     int nbthreads;
-    vpmap_thread_t **threads;
+    vpmap_thread_t* threads;
+    hwloc_cpuset_t cpuset;
 } vpmap_t;
 
-static int vpmap_nb_total_threads = -1; /**< maintains the total number of threads */
-
-static vpmap_t *map = NULL;     /**< used by the from_file and from_affinity only */
-static int nbvp = -1;
-static int nbht = 1;
-static int nbthreadspervp = -1; /**< used by the from_parameters only */
-static int nbcores = -1;        /**< used by the from_parameters only */
-
-static int vpmap_get_nb_threads_in_vp_parameters(int vp);
-static int vpmap_get_nb_cores_affinity_parameters(int vp, int thread);
-static void vpmap_get_core_affinity_parameters(int vp, int thread, int *cores, int *ht);
-
+static vpmap_t *parsec_vpmap = NULL;     /**< used by the from_file and from_affinity only */
+static int parsec_nbvp = -1;
+static int parsec_nbht = 1;
+static int parsec_nb_total_threads = 0;
 static int parse_binding_parameter(int vp, int nbth, char * binding);
 
-int vpmap_get_nb_total_threads(void)
```

Comment:
This is used by dplasma, and is removed without replacement. The new `parsec_context_query(parsec_context_t*)` could be used instead but we don't have the parsec context at the caller site. I substituted with `parsec_vpmap_get_vp_thread(0)` in dplasma as that was probably the intent anyway (assuming vps have symmetrical number of threads). Should we reintroduce the function?